src/guide/typescript-support.md

```diff
@@ -133,6 +133,47 @@ const Component = defineComponent({
 })
 ```
 
+### Augmenting Types for Use with Plugins or User Definitions
+
+Vue 3 provides `globalProperties` property for [plugins](./plugins.html#writing-a-plugin) to add to Vue’s global properties in each component instance. In some cases, we also need to [do it by ourselves](../api/application-config.html#globalproperties):
+
+```typescript
+// User Definition
+import axios from 'axios'
+
+const app = Vue.createApp()
+app.config.globalProperties.$http = axios
+
+// Plugin for validate some data
+export default {
+  install(app, options) {
+    app.config.globalProperties.$validate = (data: object, rule: object) => {
+      // check whether the object meets certain rules
+    }
+  }
+}
+```
+
+In order to let TypeScript know the attributes we (or plugins) added and their types, the [Module Augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation) feature can help us.
+
+For the above example, we (or plugins) can add the following code of type definition:
+
+```typescript
+import axios from 'axios'
+
+declare module '@vue/runtime-core' {
+  export interface ComponentCustomProperties {
+    axios: typeof axios
+  }
+}
+```
+
+We should put the above code wherever it can be loaded, such as `main.js` where you define these properties, or the `*.d.ts` file in `src/typings` folder automatically loaded by TypeScript. For plugins, it locates in the location specified by the `types` property in `package.json`. 
```

Comment:
```suggestion
We should put the above code wherever it can be loaded, such as `main.ts` where you define these properties, or a `*.d.ts` file in the `src/typings` folder automatically loaded by TypeScript. For plugins, it should be placed in the location specified by the `types` property in `package.json`. 
```